tradeoff-analytics/src/main/java/com/ibm/watson/developer_cloud/tradeoff_analytics/v1/TradeoffAnalytics.java

```diff
@@ -98,17 +103,56 @@ public TradeoffAnalytics(String username, String password) {
    * @return the decision problem
    */
   public ServiceCall<Dilemma> dilemmas(final Problem problem, final Boolean generateVisualization) {
+	
+	  return dilemmas(problem, generateVisualization, null);
+  }
+  
+  /**
+   * Returns a dilemma that contains the {@link Problem} and a {@link Resolution}. The {@link Problem} contains a set of
+   * options and objectives. The {@link Resolution} contains a set of optimal options, their analytical characteristics,
+   * and by default their representation on a 2D space. <br>
+   * <br>
+   * Here is an example of how to call the Tradeoff Analytics service with a Problem:
+   *
+   * <pre>
+   * TradeoffAnalytics service = new TradeoffAnalytics();
+   * service.setUsernameAndPassword(&quot;USERNAME&quot;, &quot;PASSWORD&quot;);
+   * Problem problem = new Problem(); // create the options and objectives here
+   * Dilemma dilemma = service.dilemmas(problem, false, true);
+   * </pre>
+   *
+   * @param problem the decision problem
+   * @param generateVisualization if true the Dilemma contains information to generate visualization
+   * @param findPreferableOptions if true the Dilemma includes a refined subset of best candidate options
+   * that will most likely satisfy the greatest number of users.
+   * @return the decision problem
+   */
+  
+  public ServiceCall<Dilemma> dilemmas(
+		  final Problem problem, 
+		  final Boolean generateVisualization,
+		  final Boolean findPreferableOptions) {
     Validator.notNull(problem, "problem was not specified");
 
     final String contentJson = GsonSingleton.getGsonWithoutPrettyPrinting().toJson(problem);
 
     final RequestBuilder requestBuilder =
         RequestBuilder.post(PATH_DILEMMAS).bodyContent(contentJson, HttpMediaType.APPLICATION_JSON);
-
-    if (generateVisualization != null) {
-      requestBuilder.query(GENERATE_VISUALIZATION, generateVisualization);
+    
+    List<String> queryParams = new ArrayList<>();
+    if (generateVisualization != null)
+    {
```

Comment:
Respect the existing code style. `{` should be in the line before.

Make sure you run

```
gradle check
```